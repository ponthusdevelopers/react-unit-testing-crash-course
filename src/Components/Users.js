import React from 'react'

const Users = () => {

    return (
        <div data-testid={`user-1`}>Hi</div>
    )
}

export default Users