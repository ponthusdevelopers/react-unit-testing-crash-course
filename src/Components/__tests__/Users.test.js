import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import Users from '../Users'

/* test('Should render Users component', () => {
    render(<Users />)
    const userElement = screen.getByTestId('user-1')
    expect(userElement).toBeInTheDocument()
    expect(userElement).toHaveTextContent('Hi')
}) */

afterEach(() => {
    cleanup()
})

test('Should render illegal user', () => {
    const user = {id: 1, name: 'John', age: 15}

    render(<Users user={user} />)
    const userElement = screen.getByTestId('user-1')
    expect(userElement).toBeInTheDocument()
    expect(userElement).toHaveTextContent('John')
})

test('Should render legal user', () => {
    const user = {id: 2, name: 'Kelly', age: 20}

    render(<Users user={user} />)
    const userElement = screen.getByTestId('user-2')
    expect(userElement).toBeInTheDocument()
    expect(userElement).toHaveTextContent('Kelly')
    expect(userElement).toContainHTML('green')
})

test('Matches snapshot', () => {
    const user = {id: 1, name: 'John', age: 15}
    const tree = renderer.create(<Users user={user} />).toJSON()
    expect(tree).toMatchSnapshot()
})