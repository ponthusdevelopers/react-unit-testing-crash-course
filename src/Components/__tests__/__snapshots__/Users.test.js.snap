// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  data-testid="user-1"
>
  <h1
    style={
      Object {
        "color": "red",
      }
    }
  >
    John
  </h1>
</div>
`;
